tests/Datagrid/ListMapperTest.php

```diff
@@ -108,6 +108,71 @@ public function testAddIdentifier(): void
 
         $this->listMapper->addIdentifier($fieldDescription);
         $this->assertTrue($this->listMapper->has('fooName'));
+
+        $fieldDescription = $this->listMapper->get('fooName');
+        $this->assertTrue($fieldDescription->getOption('identifier'));
+    }
+
+    public function testAddOptionIdentifier(): void
+    {
+        $this->assertFalse($this->listMapper->has('fooName'));
+        $this->assertFalse($this->listMapper->has('barName'));
+        $this->assertFalse($this->listMapper->has('bazName'));
+
+        $this->listMapper->add('barName');
+        $this->assertNull($this->listMapper->get('barName')->getOption('identifier'));
+        $this->listMapper->add('fooName', null, ['identifier' => true]);
+        $this->assertTrue($this->listMapper->has('fooName'));
+        $this->assertTrue($this->listMapper->get('fooName')->getOption('identifier'));
+        $this->listMapper->add('bazName', null, ['identifier' => false]);
+        $this->assertTrue($this->listMapper->has('bazName'));
+        $this->assertFalse($this->listMapper->get('bazName')->getOption('identifier'));
+    }
+
+    /**
+     * @group legacy
+     *
+     * @expectedDeprecation Passing a non boolean value for "identifier" option is deprecated since 3.x and will throw an exception in 4.0.
```

Comment:
```suggestion
     * @expectedDeprecation Passing a non boolean value for the "identifier" option is deprecated since 3.x and will throw an exception in 4.0.
```